src/coreclr/src/jit/lclvars.cpp

```diff
@@ -5665,6 +5680,10 @@ void Compiler::lvaAssignVirtualFrameOffsetsToLocals()
 #ifdef TARGET_XARCH
     // On x86/amd64, the return address has already been pushed by the call instruction in the caller.
     stkOffs -= TARGET_POINTER_SIZE; // return address;
+    if (lvaRetAddrVar != BAD_VAR_NUM)
+    {
+        lvaTable[lvaRetAddrVar].lvStkOffs = stkOffs;
+    }
```

Comment:
This probably needs special handling with OSR. I'll pull down this change and figure out the right fix.